src/sentry/static/sentry/app/components/progressBar.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import Tooltip from 'app/components/tooltip';
+import {Color} from 'app/utils/theme';
+
+type Props = {
+  /**
+   * The value of the progress indicator for the determinate variant. Value between 0 and 100
+   */
+  value: number;
+  /**
+   * The color of the component.
+   */
+  progressColor?: Color;
+  /**
+   * Text to show up on a hover.
+   */
+  tooltipText?: React.ComponentProps<typeof Tooltip>['title'];
+  /**
+   * Styles applied to the component's root
+   */
+  className?: string;
+};
+
+const ProgressBar = styled(({className, tooltipText}: Props) => {
+  const content = <div className={className} />;
+
+  if (tooltipText) {
+    return <Tooltip title={tooltipText}>{content}</Tooltip>;
```

Comment:
Should callers wrap ProgressBar with a tooltip instead of complicating the ProgressBar? Having the tooltip wrap the progress bar lets the page/block component configure the tooltip as required.

Reply:
agree :)